component/module/string/string.go

```diff
@@ -114,6 +115,13 @@ func (c *Component) Handler() http.Handler {
 	fa := api.NewFlowAPI(c.ctrl, r)
 	fa.RegisterRoutes("/", r)
 
-	r.PathPrefix("/{id}/").Handler(c.ctrl.ComponentHandler())
+	r.PathPrefix("/{id}/").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Re-add the full path to ensure that nested controllers propagate
+		// requests properly.
+		r.URL.Path = path.Join(c.opts.HTTPPath, r.URL.Path)
+
+		c.ctrl.ComponentHandler().ServeHTTP(w, r)
+	})
```

Comment:
@erikbaranowski Until there's shared code for abstracting away some of the repetitive work behind using a controller in a component, you'll need to mirror this change to #3338